File: packaging/functional-languages/prolog/test/consumer.pl
:- use_module('../prolog/sshfling').

:- initialization(main, main).

main([SmokeDirectory]) :-
    sshfling:run(["--version"], 0),
    sshfling:run(["init", SmokeDirectory, "--force", "--session-seconds", "60"], 0),
    directory_file_path(SmokeDirectory, 'production/sshfling-session', SessionWrapper),
    exists_file(SessionWrapper),
    !.
main(_) :-
    halt(1).

File: packaging/functional-languages/ocaml/test/consumer.ml
let () =
  if Array.length Sys.argv <> 2 then failwith "missing smoke-project path";
  let smoke = Sys.argv.(1) in
  if Sshfling.run [ "--version" ] <> 0 then failwith "version failed";
  if
    Sshfling.run
      [ "init"; smoke; "--force"; "--session-seconds"; "60" ]
    <> 0
  then failwith "init failed";
  let wrapper = Filename.concat smoke "production/sshfling-session" in
  if not (Sys.file_exists wrapper) then failwith "session wrapper missing"
